recipes/pytorch/24.7/gh200/environments.yaml

```diff
@@ -0,0 +1,33 @@
+pytorch:
+  compiler:
+  - toolchain: gcc
+    spec: gcc@12.3
+  mpi:
+    spec: cray-mpich@8.1.30
+    gpu: cuda
+  unify: true
+  specs:
+  - aws-ofi-nccl@master
+  - cmake
+  - cuda@12
+  - cray-mpich
+  - nccl
+  - ninja
+  - protobuf@3.25.3
+  - python@3.11
+  - py-numpy
+  - py-tqdm
+  - py-lightning
+  - py-typing-extensions
+  - py-torch@2.3 +distributed +cuda cuda_arch=90 +cudnn +nccl +mpi
```

Comment:
thanks Ben!
we should probably include at least the basic torch packages torchvision torchaudio like in https://pytorch.org/get-started/locally/ as they also need to compiled with the correct cuda version (the wheels might not match)